### PR TITLE
9 10 11  header elements  02

### PR DESCRIPTION
**Pull request by bixgomez**
# Scope

- [/] Create logo atom.
- [/] Create social media atom.
- [/] Create header organism.

# Steps Completed

- [/] Created logo atom, complete with logo image and theming.
- [/] Created social media atom, complete with icons and theming.
- [/] Used cssGrid to lay out the basic (non-responsive) header.
- [/] Reverted original version to Nerdstein branding.
- [/] Exported new compiled design change

# Issues Satisfied

- [] https://github.com/nerdstein/nerdstein-design-library/issues/9
- [] https://github.com/nerdstein/nerdstein-design-library/issues/10
- [] https://github.com/nerdstein/nerdstein-design-library/issues/11

# Github Branch

- [] https://github.com/bixgomez/nerdstein-design-library/tree/9-10-11--header-elements--02

### CODE REVIEW

File: design-library/source/_patterns/01-atoms/08-logo/logo.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+  "image_url": "/images/nerdstein_11.png",
+  "image_alt": "GSA Logo",
```
Comment (nerdstein) [inline design-library/source/_patterns/01-atoms/08-logo/logo.json]: GSA?

File: design-library/source/_patterns/01-atoms/08-logo/logo.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
+# Atom: List Title
+
+*Belongs to*
+
+* Molecule: List Element
```
Comment (nerdstein) [inline design-library/source/_patterns/01-atoms/08-logo/logo.md]: I don't think this "belongs to" is correct

File: design-library/source/_patterns/01-atoms/09-social-icons/social-icons.md [hunk 3]
```diff
@@ -0,0 +1,6 @@
+# Atom: List Title
```
Comment (nerdstein) [inline design-library/source/_patterns/01-atoms/09-social-icons/social-icons.md]: Please update this file with the relevant info -- it's using "list title"

File: design-library/source/_patterns/03-organisms/02-header/header.md [hunk 4]
```diff
@@ -0,0 +1,4 @@
+---
```
Comment (nerdstein) [inline design-library/source/_patterns/03-organisms/02-header/header.md]: Please follow the conventions i used in the other patters, which captures pattern dependencies

File: design-library/source/_patterns/03-organisms/02-header/header.twig [hunk 5]
```diff
@@ -0,0 +1,8 @@
+<header id="site-header" class="grid">
+  <div class="grid-cell grid-cell--logo">
+    {% include "@atoms/logo.twig"  %}
+  </div>
+  <div class="grid-cell grid-cell--social-icons">
+    {% include "@atoms/social-icons.twig"  %}
```
Comment (nerdstein) [inline design-library/source/_patterns/03-organisms/02-header/header.twig]: so, ideally these are parameterized based on the mapping of values passed to this. The logo may be static (it's my logo), but the social icons i may want to wire into Drupal and map a list of values here. Can you please refactor this file to use the "with" utility to pass the "social media" JSON data you are already passing to the header?

**Review by nerdstein — CHANGES_REQUESTED** [review]
Please see comments

File: design-library/source/_patterns/01-atoms/08-logo/logo.json (hunk 1 above)
Comment (bixgomez) [reply design-library/source/_patterns/01-atoms/08-logo/logo.json]: Ah yes, I had chosen the GSA logo when I was trying to turn this into a USWDS library ;)

File: design-library/source/_patterns/01-atoms/08-logo/logo.md (hunk 2 above)
Comment (bixgomez) [reply design-library/source/_patterns/01-atoms/08-logo/logo.md]: Quite right!

File: design-library/source/_patterns/01-atoms/09-social-icons/social-icons.md (hunk 3 above)
Comment (bixgomez) [reply design-library/source/_patterns/01-atoms/09-social-icons/social-icons.md]: Got it -- I had been neglecting the .md files entirely!

File: design-library/source/_patterns/03-organisms/02-header/header.md (hunk 4 above)
Comment (bixgomez) [reply design-library/source/_patterns/03-organisms/02-header/header.md]: Indeed; done!

File: html/patterns/02-molecules-04-social-icons-social-icons/02-molecules-04-social-icons-social-icons.html [hunk 6]
```diff
@@ -18,22 +18,13 @@
...
-        Like us on Facebook
-      </a>
-    </li>
-      <li class="social_media--item">
-      <a href="http://www.linkedIn.com" title="Follow us on LinkedIn" class="social_media--icon LinkedIn">
-        Follow us on LinkedIn
-      </a>
-    </li>
-  </ul>
+      <div class="section-content">
```
Comment (nerdstein) [inline html/patterns/02-molecules-04-social-icons-social-icons/02-molecules-04-social-icons-social-icons.html]: hmmm its weird to see this markup here. do you have a sense of why "section content" is showing up in social icons?

File: html/patterns/03-organisms-02-header-header/03-organisms-02-header-header.html [hunk 7]
```diff
@@ -23,29 +23,15 @@
...
-    </li>
-      <li class="social_media--item">
-      <a href="http://www.linkedIn.com" title="Follow us on LinkedIn" class="social_media--icon LinkedIn">
-        Follow us on LinkedIn
-      </a>
-    </li>
-  </ul>
+
+    
+    @atoms/social-icons.twig
```
Comment (nerdstein) [inline html/patterns/03-organisms-02-header-header/03-organisms-02-header-header.html]: This seems to be an error in processing

**Review by nerdstein — COMMENTED** [review]
I'm not clear on why the html export looks funky --- see my comments. Maybe it needs exported again?

**Comment by bixgomez** [comment]
UPDATED PR SUBMITTED

https://github.com/nerdstein/nerdstein-design-library/pull/32

**Pull request closed without merging**